File: VB/RangeDate/MyGridControl/MyOptionsColumnFilter.vb
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.Utils.Controls

Namespace DateRange

    Public Class MyOptionsColumnFilter
        Inherits OptionsColumnFilter

        Protected Friend UseFilterPopupRangeDateMode As Boolean = False

        Private filterPopupModeField As FilterPopupModeExtended

        Public Overloads Property FilterPopupMode As FilterPopupModeExtended
            Get
                Return filterPopupModeField
            End Get

            Set(ByVal value As FilterPopupModeExtended)
                If FilterPopupMode = value Then Return
                Dim prevValue As FilterPopupModeExtended = FilterPopupMode
                filterPopupModeField = value
                OnChanged(New BaseOptionChangedEventArgs("FilterPopupMode", prevValue, FilterPopupMode))
            End Set
        End Property
    End Class
End Namespace

File: CS/RangeDate/MyGridControl/MyOptionsColumnFilter.cs
using System;
using System.Linq;
using DevExpress.XtraGrid.Columns;
using System.Collections.Generic;
using DevExpress.Utils.Controls;

namespace DateRange {
    public class MyOptionsColumnFilter : OptionsColumnFilter {
        protected internal bool UseFilterPopupRangeDateMode = false;
        FilterPopupModeExtended filterPopupMode;

        public MyOptionsColumnFilter(GridColumn column) : base(column)
        {
        }

        public new FilterPopupModeExtended FilterPopupMode {
            get { return filterPopupMode; }
            set {
                if(FilterPopupMode == value) return;
                FilterPopupModeExtended prevValue = FilterPopupMode;
                filterPopupMode = value;
                OnChanged(new BaseOptionChangedEventArgs("FilterPopupMode", prevValue, FilterPopupMode));
            }
        }
    }
}
